pymeasure/instruments/lecroy/lecroyT3DSO1204.py

```diff
@@ -142,6 +148,40 @@ def _math_define_validator(value, values):
     return output
 
 
+class _ChunkResizer:
+    """The only purpose of this class is to resize the chunk size of the instrument adapter.
+    This is necessary when reading a big chunk of data from the oscilloscope like image dumps and
+    waveforms.
+    Note.
+    Only if the new chunk size is bigger than the current chunk size, it is resized. """
+
+    def __init__(self, adapter, chunk_size):
+        """ Just initialize the object attributes.
+        :param: adapter of the instrument. This is usually accessed through the
+        Instrument::adapter attribute.
+        :chunk_size: new chunk size (int).
+        """
+        self.adapter = adapter
+        self.old_chunk_size = None
+        self.new_chunk_size = int(chunk_size)
+
+    def __enter__(self):
+        """ Only resize the chunk size if the adapter support this feature"""
+        if (hasattr(self.adapter, "connection")
```

Comment:
`isinstance(adapter, Adapter)` is maybe more idiomatic/clear (although both come out to the same thing).

Reply:
I'm not sure, whether the Protocol Adapter has a connection.

Reply:
I had to add that check because when mocking the adapter in unit testing through `expected_protocol`, I was getting an expection about the connection attribute not being found. I guess this is what Benedikt is hinting to.

Reply:
As the `Adapter` class already has the attribute (albeit set to None), any Adapter should have that attribute. I'd test whether the adapter is an instance of VISAAdapter, that way you ensure, that it has a connection chunk_size.

Reply:
Good, defensive solution implemented.